### ISSUE TITLE
Results of Single Subject Personalization Seem Not Good

### ISSUE THREAD

**Issue by vitrun** [fenghora/personalize-anything#8]
Hi, 
Thanks for sharing your promising research. I tried experimenting with the `single_subject_personalization.ipynb` from the repo, but did not get results as good as those shown in the paper.

In the first attempt, I ran the notebook as-is and got this:

![Image](https://github.com/user-attachments/assets/244f0d69-eb61-44aa-af1a-159c6afe060a)

I noticed there is a trailing space in the prompt:
```python
key_word = "teddy bear "
```
Also there are two directories in the `example_data` for the bear `teddy_bear` and `teddy_bear_`, each with different masks. So I removed the training space, but the result was still not great:

![Image](https://github.com/user-attachments/assets/9c96bf18-a34d-4df2-9f97-bebe10e20490)

For the cat example, the result was better:

![Image](https://github.com/user-attachments/assets/85245122-8ab0-4842-9d73-5cc733241c69)

However, all the result share the same issue: the generated results are heavily constrained by the mask, with no variations in poses or gestures which make it more like copy-paste rathar than true personalization.

To address this, I replaced the default mask for the teddy mask with this one: 

![Image](https://github.com/user-attachments/assets/d632b1a3-2789-4270-a20a-8df658433ae7)

Unfortunately, the result was still not satisfactory:

![Image](https://github.com/user-attachments/assets/6d885451-b135-44ed-ae7f-c2dc4ca5793c)

Any suggestions on how to improve this?

**Comment by vitrun** [comment]
BTW. My environment is: 
- hardware: A100
- python libs:
   - torch==2.6.0
   - transformers==4.49.0
   - diffusers==0.32.0

**Comment by huanngzh** [comment]
Hi, so sorry for our mistakes. We have fixed our code and updated some instructions in our notebook.

Some instructions:
* The mask in the code refers to the region of the subject in the reference image, which is used to extract the specified tokens from the reference image tokens.
* We show an example of manually perturbing the mask for personalization. Our automatic perturbation code will be released if the paper is accepted.

Another tip: The results of our method may produce non-optimal results due to the instability of inversion and the original distribution of pretrained models. If you are not satisfied with the results, you can generate them several times and pick the most satisfactory result.

**Comment by vitrun** [comment]
Thanks for your reply.

> * We show an example of manually perturbing the mask for personalization

Are you referring to the bear mask with the hand cropped out? That level of pertubation seems insufficient to enable diverse poses or gestures in the generated results. Are the following examples from the paper a result of such perturbation?

![Image](https://github.com/user-attachments/assets/7f87f42c-db55-488c-b762-ca1b3b683062)

**Comment by vitrun** [comment]
Referring to the eq 2 from the paper:
> \hat{X} = X \odot (1-M) + X_{ref} \odot M
Here, `M` can differ from the `M_{ref}`, it does accept a custom mask to specify the subject's location in the generated image. However, it is not clearly explained what it would be if the custom mask represents a different segmentation. I believe this is important and would be great if the paper includes experiments exploring this scenario.

**Comment by cryptexis** [comment]
Hi @huanngzh thanks for a great work.

I have read the paper, went through examples and through the issues in the repo. And this seems to be the most relevant place to write.

In the image @vitrun has presented. The mask `M'` will be quite differently than the mask `M` in the target image. I would assume for rigid objects like a can of soda ("including simulating natural shape variations using morphological dilation/erosion with a 5px kernel") mask perturbations can work. But for the dragon it seems a bit unlikely. 

Would it be possible to share the code before the paper acceptance?

**Comment by Mega-wisdom** [comment]
> Hi [@huanngzh](https://github.com/huanngzh) thanks for a great work.
> 
> I have read the paper, went through examples and through the issues in the repo. And this seems to be the most relevant place to write.
> 
> In the image [@vitrun](https://github.com/vitrun) has presented. The mask `M'` will be quite differently than the mask `M` in the target image. I would assume for rigid objects like a can of soda ("including simulating natural shape variations using morphological dilation/erosion with a 5px kernel") mask perturbations can work. But for the dragon it seems a bit unlikely.
> 
> Would it be possible to share the code before the paper acceptance?

yeah, I'm agree with you. The mask M of the original image and the mask M of the target image are completely different. For the case in the paper, when using official code, I am unable to generate images that perform as well as in the paper. In fact, there will be serious artifacts in the generated images, just like the example of the teddy bear shown above.
